### DISCUSSION TITLE
model high speed bullet using openradioss

### DISCUSSION THREAD

**Discussion by wanglei742** [OpenRadioss/OpenRadioss#2395] [Q&A]
i saw an animation using the CFD to model the bullet, so i believe the openradioss can also replicate this phenomenon. but I face some issues below, can anyone give me some cues or guidance? very much appreciated for this,pls see attached
![Snipaste_2024-03-06_12-46-10](https://github.com/OpenRadioss/OpenRadioss/assets/99164098/5f937a97-10f6-467d-9c12-55e37e87b457)
the attached is the axisysm model and model size is very small, before running the attached pls change the .txt to .rad 
[bullet_0000.txt](https://github.com/OpenRadioss/OpenRadioss/files/14504710/bullet_0000.txt)
[bullet_0001.txt](https://github.com/OpenRadioss/OpenRadioss/files/14504711/bullet_0001.txt)
Thanks

**Comment by RevolT87** [comment]
Hi @wanglei742,
This is indeed possible to do this kind of simulation using OpenRadioss. You need to use the multimaterial law 151. Here is an example of an air burst model: 
https://community.altair.com/csm?id=altair_file_exchange_article_view&sysparm_article=KB0119695 
You can use the 2D axi-symmetrical model as a source of insipiration.
Regards,
Mathis.

**Reply by wanglei742** [reply]
Hi, Mathis, thanks for your reply and response, one more question is can we model the 'solid' part in axi-sym problem? the link you provided both TNT and air are modeled with fluid property, i thought the bullet should be modeled with solid property. but i didn't find a right material model for the bullet part. 
very much appreciated for your reply.

**Reply by RevolT87** [reply]
Hi, 
When I was doing this kind of study, I had to consider the bullet as a fluid. I was using Noble-Able EoS for the bullet. 
obviously the bullet will be difusive, but thé pressure patern was reallistic.

**Reply by Schwoertzig** [reply]
Hi @wanglei742,

Additionally, here are some supplementary comments.

In the context of finite element analysis, "solid property" refers to elements such as hexahedral (hexa) and tetrahedral (tetra). These elements can be utilized to model both fluid and solid states. It's advisable to consistently employ /PROP/SOLID, as it comes with default pseudo-viscosity parameters (qa, qb) beneficial for shock wave modeling and ensuring stability in staggered schemes. Note that all material laws except for Law 151, which relies on a colocated scheme, benefit from this default setting. Hence, it's prudent to refrain from using /prop/fluid unless you specifically intend to deactivate pseudo viscosity in staggered schemes.

As of now, Material Law 151 isn't compatible with solid materials; it's designed to simulate fluid behavior (where deviatoric stress equals zero)  with high velocity, though this doesn't preclude the possibility of obtaining negative pressures. To achieve tension behavior in such elements, you may need to adjust the bullet equation of state (EoS) accordingly.

Moving forward, there are two potential avenues for improvement in OpenRadioss for those willing to tackle the challenge:
- Ensuring compatibility with solid states, including the calculation of deviatoric stress in HLLC colocated scheme.
- Developing a sliding interface for fluid-structure interaction. While there's an existing interface (TYPE12) that could serve as a starting point, it would involve meshing two parallel sliding domains, with one defining the bullet as Lagrangian. The Lagrangian framework for the bullet will prevent diffusion and enable a wide range of constitutive laws, including modeling rigid materials.

**Reply by wanglei742** [reply]
Hi, Schwoertzig: 
thanks for your detailing explanation. let me know more behind the theory, even previously i know the hex and treta also need to model with solid prop even this should be acted as fluid behavior. very much appreciated letting me know more from your reply. 
As i am also an Altair official user, does the Radioss already has the function to meet my purpose? that is **meshing two parallel sliding domains**, seems i have never seen a example before. it seems the CFD always using this methodology to model such phenomena. 
so can you share a link or file to me for reference  if you have?
Thanks

**Reply by Schwoertzig** [reply]
@wanglei742 ,
there is an old implementation of such sliding interface for : 3D staggered scheme and mono-material law only.
( https://help.altair.com/hwsolvers/rad/topics/solvers/rad/inter_type12_starter_r.htm#inter_type12_starter_r )
What would be interesting in your case (if precision is still not enough using all recommendations brought by @RevolT87 ) would be to introduce in source code compatibility with 2D, colocated scheme, and multimaterial. 
Regards

**Reply by wanglei742** [reply]
Hi, Thanks to both, current methodology is enough for my simulation. i will try both of your advice

**Reply by wanglei742** [reply]
Hi, Schwoertzig:
hello again, I revisited your reply when i came across the solid/fluid prop 
the previous reply"**it's prudent to refrain from using /prop/fluid unless you specifically intend to deactivate pseudo viscosity in staggered schemes"**
so my question is when I need to employ the fluid prop, i still confused about the this. here is a example link in the altair website 
https://community.altair.com/community/en/mok-fluid-structure-interaction-fsi-benchmark-simulation-with-radioss?id=kb_article&sys_id=67a8910497554610e3b0361e6253afd7
you can visualize the mesh can move and deformed, is this reason the example use the solid fluid prop?
appreciated if you can reply

**Reply by Schwoertzig** [reply]
Hi @wanglei742 
You never need to utilize /PROP/FLUID. You can simply disregard it. It's essentially a shortcut to /PROP/SOLID with qa=qb=0.0, which is not advisable.
Regards

**Reply by wanglei742** [reply]
Hi, Schwoertzig:
based on your reply, let me just forget this fluid prop. 
very much thanks
